test/extensions/transport_sockets/internal_upstream/internal_upstream_integration_test.cc

```diff
@@ -182,5 +191,392 @@ TEST_F(InternalUpstreamIntegrationTest, BasicFlowWithoutTransportSocket) {
+    /*
+    auto* tls_params = common_tls_context->mutable_tls_params();
+    tls_params->set_tls_minimum_protocol_version(
+      envoy::extensions::transport_sockets::tls::v3::TlsParameters::TLSv1_0);
+    tls_params->set_tls_maximum_protocol_version(
+      envoy::extensions::transport_sockets::tls::v3::TlsParameters::TLSv1_3);
+    */
+    auto* tls_certificate = common_tls_context->add_tls_certificates();
+    tls_certificate->mutable_certificate_chain()->set_filename(TestEnvironment::runfilesPath("test/config/integration/certs/servercert.pem"));
+    tls_certificate->mutable_private_key()->set_filename(TestEnvironment::runfilesPath("test/config/integration/certs/serverkey.pem"));
+    transport_socket->mutable_typed_config()->PackFrom(tls_context);
+  }
+}
+
+// Setup the external listener which listens to the client messages.
+void HttpsInspectionIntegrationTest::setupExternalListenerConfig() {
+  config_helper_.addConfigModifier([&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    auto* static_resources = bootstrap.mutable_static_resources();
+    auto* listener = static_resources->mutable_listeners(0);
+    envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager hcm;
+    hcm.set_stat_prefix("ingress_http");
+    auto * route_config = hcm.mutable_route_config();
+    route_config->set_name("local_route");
+    auto * virtual_hosts = route_config->add_virtual_hosts();
+    virtual_hosts->set_name("local_service");
+    virtual_hosts->add_domains("*");
+    auto * routes = virtual_hosts->add_routes();
+    routes->mutable_match()->set_prefix("/");
+    routes->mutable_route()->set_cluster("internal_upstream");
+
+    // Setup router HTTP filter.
+    auto* router_filter = hcm.add_http_filters();
+    router_filter->set_name("envoy.filters.http.router");
+    envoy::extensions::filters::http::router::v3::Router router;
+    router_filter->mutable_typed_config()->PackFrom(router);
+    ConfigHelper::setConnectConfig(hcm, true, false);
+    // Setup listener filter chain.
+    auto* filter_chain = listener->mutable_filter_chains(0);
+    // Clear existing http proxy filters.
+    filter_chain->clear_filters();
+    auto* filter = filter_chain->add_filters();
+    filter->set_name("envoy.filters.network.http_connection_manager");
+    filter->mutable_typed_config()->PackFrom(hcm);
+  });
+}
+
+// Setup the internal upstream cluster which bridge the traffic back to internal listener.
+void HttpsInspectionIntegrationTest::setupInternalUpstreamConfig() {
+  config_helper_.addConfigModifier([&](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+    auto* static_resources = bootstrap.mutable_static_resources();
+    auto* cluster = static_resources->mutable_clusters()->Add();
+    cluster->set_name("internal_upstream");
+    // Insert internal upstream transport.
+    TestUtility::loadFromYaml(R"EOF(
+        name: envoy.transport_sockets.internal_upstream
```

Comment:
Do you really need to pass metadata for this test? I think it would be good to keep test contained to the data path first, before making it more complicated.

Reply:
done